conf.c

```diff
@@ -128,17 +129,34 @@ const char *conf_ifname(void)
 /* Return full path of rcfile. Allocates string which must bee free()-d. */
 static char *get_confname(void)
 {
-	char *full_path, *homedir = getenv("HOME");
+	char *full_path;
+	char *homedir = getenv("HOME");
+	char *xdg_env = getenv("XDG_CONFIG_HOME");
+	char *xdg_config_dir;
 	struct passwd *pw;
 
-	if (homedir == NULL) {
+
+	if (homedir == NULL && xdg_env == NULL) {
 		pw = getpwuid(getuid());
 		if (pw == NULL)
 			err_quit("can not determine $HOME");
 		homedir = pw->pw_dir;
 	}
-	full_path = malloc(strlen(homedir) + strlen(CFNAME) + 3);
-	sprintf(full_path, "%s/%s", homedir, CFNAME);
+
+	// use XDG_CONFIG_HOME/wavemon if available
+	xdg_config_dir = malloc(strlen(xdg_env) + strlen(NAME) + 2);
+	sprintf(xdg_config_dir, "%s/%s", xdg_env, NAME);
+	DIR* check_xdgdir = opendir(xdg_config_dir);
```

Comment:
Can you use a `stat(2)` and check if the return value (easier, since otherwise it leaks an opened dirfd).
```c
struct stat st;

if (stat(xdg_config_dir, &st) == 0) {
     full_path = malloc(/*... */);
// ...
```
Could test for `S_IFDIR`, but unlikely to have a regular file at this place (and failure would be evident later).